src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +25,34 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws FileNotFoundException {
+        String line = "";
+        TreeSet<Character> treeSet = new TreeSet<>();
+        try (BufferedReader fin = new BufferedReader(new FileReader(new File(fileName)))) {
+            while ((line = fin.readLine()) != null) {
+                line = line.replaceAll("[0-9]*\\W*", "");
+                line = line.toLowerCase();
+                char[] chars = line.toCharArray();
+                for (int i = 0; i < line.length(); i++) {
+                    treeSet.add(chars[i]);
+                    System.out.println(treeSet);
+                }
+            }
+        } catch (IOException e) {
+            throw new FileNotFoundException();
+        }
+        StringBuilder stringBuilder = new StringBuilder();
+        Iterator<Character> itr = treeSet.iterator();
+        while (itr.hasNext()) {
+            stringBuilder.append(itr.next());
+        }
+        if (stringBuilder.length() == 0) {
+            return "";
+        }
+        if (stringBuilder.length() < 5) {
+            return stringBuilder.toString();
+        }
+        return stringBuilder.delete(5, stringBuilder.length()).toString();
```

Comment:
lets add to builder not more than 5 chars not to delete them later

Reply:
Done